### PR TITLE
Add Type CLI Commands

### PR DESCRIPTION
**Pull request by james-finkle**
Signed-off-by: James Finkle <james.finkle@prominentedge.com>

### CODE REVIEW

File: core/store/src/main/java/org/locationtech/geowave/core/store/cli/type/DescribeTypeCommand.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    final StoreLoader inputStoreLoader = new StoreLoader(inputStoreName);
+    if (!inputStoreLoader.loadFromConfig(configFile)) {
+      throw new ParameterException("Cannot find store name: " + inputStoreLoader.getStoreName());
+    }
+    inputStoreOptions = inputStoreLoader.getDataStorePlugin();
+
+    LOGGER.info(
+        "Describing everything in store: " + inputStoreName + " with type name: " + typeName);
+    DataTypeAdapter[] types = inputStoreOptions.createDataStore().getTypes();
```
Comment (jdgarrett) [inline core/store/src/main/java/org/locationtech/geowave/core/store/cli/type/DescribeTypeCommand.java]: There is a `getType` function on DataStore that can get an adapter by type name.  That should eliminate the need for the loop and the check.

File: core/store/src/main/java/org/locationtech/geowave/core/store/cli/type/ListTypesCommand.java [hunk 2]
```diff
@@ -6,20 +6,27 @@
  * under the terms of the Apache License, Version 2.0 which accompanies this distribution and is
  * available at http://www.apache.org/licenses/LICENSE-2.0.txt
  */
-package org.locationtech.geowave.core.store.cli.store;
+package org.locationtech.geowave.core.store.cli.type;
 
 import java.io.File;
+import java.io.IOException;
```
Comment (jdgarrett) [inline core/store/src/main/java/org/locationtech/geowave/core/store/cli/type/ListTypesCommand.java]: Clean up unused imports.

File: core/store/src/main/java/org/locationtech/geowave/core/store/cli/type/RemoveTypeCommand.java [hunk 3]
```diff
@@ -6,22 +6,26 @@
...
  * available at http://www.apache.org/licenses/LICENSE-2.0.txt
  */
-package org.locationtech.geowave.core.store.cli.store;
+package org.locationtech.geowave.core.store.cli.type;
 
 import java.io.File;
 import java.util.ArrayList;
 import java.util.List;
 import org.locationtech.geowave.core.cli.annotations.GeowaveOperation;
+import org.locationtech.geowave.core.cli.api.Command;
```
Comment (jdgarrett) [inline core/store/src/main/java/org/locationtech/geowave/core/store/cli/type/RemoveTypeCommand.java]: Clean up unused imports.

File: core/store/src/main/java/org/locationtech/geowave/core/store/cli/type/TypeSection.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+package org.locationtech.geowave.core.store.cli.type;
+
+import org.locationtech.geowave.core.cli.annotations.GeowaveOperation;
+import org.locationtech.geowave.core.cli.api.DefaultOperation;
+import org.locationtech.geowave.core.cli.operations.GeowaveTopLevelSection;
+import com.beust.jcommander.Parameters;
+
+@GeowaveOperation(name = "type", parentOperation = GeowaveTopLevelSection.class)
+@Parameters(
+    commandDescription = "Commands to list, remove, and describe data types within data stores")
```
Comment (jdgarrett) [inline core/store/src/main/java/org/locationtech/geowave/core/store/cli/type/TypeSection.java]: Maybe change to "Commands for managing types within a data store"

File: services/api/src/main/java/org/locationtech/geowave/service/TypeService.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
+package org.locationtech.geowave.service;
```
Comment (jdgarrett) [inline services/api/src/main/java/org/locationtech/geowave/service/TypeService.java]: Missing copyright header.

File: services/client/src/main/java/org/locationtech/geowave/service/client/TypeServiceClient.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
+package org.locationtech.geowave.service.client;
```
Comment (jdgarrett) [inline services/client/src/main/java/org/locationtech/geowave/service/client/TypeServiceClient.java]: Missing copyright header.

File: test/src/test/java/org/locationtech/geowave/test/services/TypeServicesIT.java [hunk 7]
```diff
@@ -0,0 +1,156 @@
+package org.locationtech.geowave.test.services;
```
Comment (jdgarrett) [inline test/src/test/java/org/locationtech/geowave/test/services/TypeServicesIT.java]: Missing copyright header.

File: core/store/src/main/java/org/locationtech/geowave/core/store/cli/type/DescribeTypeCommand.java (hunk 1 above)
Comment (james-finkle) [reply core/store/src/main/java/org/locationtech/geowave/core/store/cli/type/DescribeTypeCommand.java]: getType does not yet exist on master. Skipping this change for now.

**Pull request merged**